functions/scl_packages.pp

```diff
@@ -0,0 +1,19 @@
+# @summary Generate package names that may or may not be in some SCL
+# @param packages
+#   The package names without any SCL prefix
+# @param scl
+#   An optional SCL name, without a dash as suffix
+# @return The package names, with SCL prefix if applicable
+#
+# @example Without SCL
+#   katello_devel::scl_packages(['postgresql-server']) == ['postgresql-server']
+#
+# @example With SCL
+#   katello_devel::scl_packages(['postgresql-server'], 'rh-postgresql12') == ['rh-postgresql12-postgresql-server']
+function katello_devel::scl_packages(Array[String[1]] $packages, Optional[String[1]] $scl = undef) >> Array[String[1]] {
```

Comment:
Since this can return non-SCL packages, I think the function should be called `packages` as it gives a false impression when you read it's use in code. Something like (https://github.com/theforeman/puppet-katello_devel/pull/245/files#diff-29fdc8c170b6278c50ae7a6359737e386dc4769ce94b9a6e505f5bcd3e4890deR26-R34) is repetitive in a sense, but it is also incredibly clear and easy to read what is happening for a developer.